profiling/src/profile/api.rs

```diff
@@ -240,32 +226,29 @@ fn function_fetch(pprof: &pprof::Profile, id: u64) -> anyhow::Result<Function> {
     }
 }
 
-fn lines_fetch<'a>(
-    pprof: &'a pprof::Profile,
-    lines: &'a [pprof::Line],
-) -> anyhow::Result<Vec<Line<'a>>> {
-    let mut output = Vec::with_capacity(lines.len());
-    for line in lines {
-        output.push(Line {
-            function: function_fetch(pprof, line.function_id)?,
-            line: line.line,
-        });
-    }
-    Ok(output)
-}
-
 fn location_fetch(pprof: &pprof::Profile, id: u64) -> anyhow::Result<Location> {
     if id == 0 {
         return Ok(Location::default());
     }
 
     match pprof.locations.iter().find(|item| item.id == id) {
-        Some(location) => Ok(Location {
-            mapping: mapping_fetch(pprof, location.mapping_id)?,
-            address: location.address,
-            lines: lines_fetch(pprof, &location.lines)?,
-            is_folded: location.is_folded,
-        }),
+        Some(location) => {
+            anyhow::ensure!(!location.is_folded, "expected Location to not be folded");
+            anyhow::ensure!(
+                location.lines.len() == 1,
```

Comment:
I noticed that this is actually incorrect, although it has been incorrect for a long time.

Looking at the pprof format ( https://github.com/DataDog/libdatadog/blob/a351844ed09611d01e853c9cf476291f519bc9e7/profiling/src/profile/profile.proto#L181 ) or even [Felix's great pic](https://github.com/DataDog/go-profiler-notes/blob/main/pprof.md): 

![https://github.com/DataDog/go-profiler-notes/blob/main/pprof.md](https://github.com/DataDog/go-profiler-notes/raw/main/profile.png) 

...there's no `lines` entry inside pprof (it's `line`). Our `pprof.rs` has incorrectly diverged from the `pprof` definition.

I'm guessing this hasn't caused issues because pprof doesn't encode the field names (?), so because we added at the same location the same field but with a different name, I think the resulting pprof is still equivalent.

So... we can probably leave it to a separate PR but it's worth maybe going back to the upstream `profile.proto` and make sure our copy is 1:1, and then pprof.rs is also 1:1 with it :)

Reply:
I think `Location.line` does exist, and I just named it `lines`. The name in our source code doesn't matter, as long as the type and order are the same.

Reply:
Yeah, definitely ok to not change it in this PR.

> But, if you want to have exactly the same names in our source code as in the pprof

Is it too nitpicky? It seems... weird to me that in the code that is implementing pprof we're not using the names that pprof uses.

It was definitely confusing to me -- I was referencing the pprof drawing above when understanding the code and the different name made me think I was misunderstanding the code and that I was looking at our own internal representation (which can use `lines`) vs the pprof one (which I was sure didn't use `lines`). I even wondered if perhaps you had made a typo and was the code compiling (maybe it wasn't?)? So I checked it out locally, verified it compiled, checked my assumptions, and that's when I found the name was changed.

So... Yeah, I was really led astray by the off name, hence why I suggested fixing it.